api/services/v1alpha1/auth_types.go

```diff
@@ -39,6 +39,10 @@ type AuthSpec struct {
 	// AllowedGroups cannot contain empty strings, but 'system:authenticated' is allowed for general access
 	// +kubebuilder:validation:XValidation:rule="self.all(group, group != '')",message="AllowedGroups cannot contain empty strings"
 	AllowedGroups []string `json:"allowedGroups"`
+	// MetricsAdminGroups defines groups with administrative access to metrics resources. Cannot contain 'system:authenticated' or empty strings
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self.all(group, group != 'system:authenticated' && group != '')",message="MetricsAdminGroups cannot contain 'system:authenticated' or empty strings"
+	MetricsAdminGroups []string `json:"metricsAdminGroups,omitempty"`
```

Comment:
Why are we adding a new group rather than just giving the metrics role to the adminGroup already defined? Would we expect a metricsAdmin to be different from an admin?

Most users are expected to manage the admin and allowed groups through the dashboard UI so this would be different.

Reply:
cluster-admins still have this privilege, as for why i'm creating a separate group it's just to follow the principle of least privilege, this way you can still have people managing metrics that are not admins